src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc

```diff
@@ -400,33 +400,29 @@ static void pf_connectivity_changed_locked(void* arg, grpc_error* error) {
       p->latest_pending_subchannel_list = nullptr;
       grpc_connectivity_state_set(
           &p->state_tracker, GRPC_CHANNEL_TRANSIENT_FAILURE,
-          GRPC_ERROR_REF(error), "selected_not_ready+switch_to_update");
+          GRPC_ERROR_REF(error), "selected_not_ready+switch_to_update", false);
     } else {
-      // TODO(juanlishen): we re-resolve when the selected subchannel goes to
-      // TRANSIENT_FAILURE because we used to shut down in this case before
-      // re-resolution is introduced. But we need to investigate whether we
-      // really want to take any action instead of waiting for the selected
-      // subchannel reconnecting.
-      GPR_ASSERT(sd->curr_connectivity_state != GRPC_CHANNEL_SHUTDOWN);
-      if (sd->curr_connectivity_state == GRPC_CHANNEL_TRANSIENT_FAILURE) {
-        // If the selected channel goes bad, request a re-resolution.
+      // A subchannel can be in TRANSIENT_FAILURE only before it's connected.
+      GPR_ASSERT(sd->curr_connectivity_state != GRPC_CHANNEL_TRANSIENT_FAILURE);
```

Comment:
Before this PR, the subchannel will report `TRANSIENT_FAILURE` both when it fails to connect or disconnects. Now it reports `TRANSIENT_FAILURE` when it fails to connect, and `SHUTDOWN` when it disconnects. 

But it doesn't affect the behavior of this code block. 

Before this PR, the selected subchannel will only be in `TRANSIENT_FAILURE` if it disconnects. Now, the selected subchannel will only be in `SHUTDOWN` if it disconnects. Either way, disconnection is the only condition for PF to re-resolve.